### PR TITLE
AP_EFI: use enum class for protocol

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Review by WickedShell — COMMENTED** [review]
Does actually make the build larger. Have you checked if making the enum class specified as being stored in a `int8_t` keeps the size the same? (Basically means no real casting on the comparisons to the params)

File: libraries/AP_EFI/AP_EFI.h [hunk 1]
```diff
@@ -60,18 +60,18 @@ class AP_EFI {
...
 
     // Parameter info
     static const struct AP_Param::GroupInfo var_info[];
 
     // Backend driver types
-    enum EFI_Communication_Type {
-        EFI_COMMUNICATION_TYPE_NONE      = 0,
-        EFI_COMMUNICATION_TYPE_SERIAL_MS = 1,
-        EFI_COMMUNICATION_TYPE_NWPMU     = 2,
+    enum class Type {
```
Comment (tridge) [inline libraries/AP_EFI/AP_EFI.h:71]: should be uint8_t size

**Review by tridge — CHANGES_REQUESTED** [review]
should use AP_Enum I think

File: libraries/AP_EFI/AP_EFI.h (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_EFI/AP_EFI.h]: Done.

**Comment by peterbarker** [comment]
> should use AP_Enum I think

Done

**Review by tridge — APPROVED** [review]

**Comment by WickedShell** [comment]
Still larger :)

Out of curiosity swapping from AP_Int8 to AP_Enum is lossless for anyone who already set it? What's the advantage of AP_Enum at that point?

**Pull request merged**